src/Conf.ml

```diff
@@ -993,20 +1055,50 @@ let kind_of fname =
   | ".mlt" -> `Use_file
   | _ -> !kind
 
-let update_using_xdg =
+let xdg_config =
   match Caml.Sys.getenv_opt "XDG_CONFIG_HOME" with
   | Some xdg_config_home ->
       let filename =
         Filename.concat xdg_config_home "ocamlformat/.ocamlformat"
       in
-      Staged.stage (fun conf -> read_config ~filename ~parents:false conf)
-  | None -> Staged.stage (fun conf -> conf)
+      if Caml.Sys.file_exists filename then Some filename else None
+  | None -> None
 
 let build_config ~filename =
-  default
-  |> Staged.unstage update_using_xdg
-  |> read_config ~filename ~parents:true
-  |> update_using_env |> C.update_using_cmdline
+  let files, project_root =
+    collect_files ~dir:(Filename.dirname (to_absolute filename)) []
+  in
+  let files =
+    match (xdg_config, !inside_project_only) with
+    | None, _ | Some _, true -> files
+    | Some f, false -> f :: files
+  in
+  let verbose = !print_config in
+  if verbose then
+    Option.iter project_root ~f:(Format.eprintf "project-root=%s\n%!") ;
```

Comment:
call `to_relative`?

Reply:
I don't think so.
It would be harder to read `project-root=../../../../`